app/models/category.rb

```diff
@@ -7,7 +7,14 @@ class Category < ApplicationRecord
 
   has_many :children, class_name: "Category", inverse_of: :parent
   belongs_to :parent, class_name: "Category", optional: true
-  has_and_belongs_to_many :properties
+
+  has_and_belongs_to_many :properties,
+    join_table: :categories_properties,
+    foreign_key: :category_id,
+    association_foreign_key: :property_friendly_id
+  def property_friendly_ids=(ids)
+    self.properties = Property.where(friendly_id: ids)
+  end
```

Comment:
I don't think this relationship is set up correctly. When I tried running it on your branch, the attributes array is empty for all the categories in categories.json

Reply:
I imagine the other relationships will have similar issue, that's just the one I checked